compiler/rustc_infer/src/infer/error_reporting/nice_region_error/util.rs

```diff
@@ -64,7 +64,7 @@ pub fn find_param_with_region<'tcx>(
     let body_id = hir.maybe_body_owned_by(def_id)?;
 
     let owner_id = hir.body_owner(body_id);
-    let fn_decl = hir.fn_decl_by_hir_id(owner_id).unwrap();
+    let fn_decl = hir.fn_decl_by_hir_id(owner_id)?;
```

Comment:
At this point in time we might also have parametrized const item, not just a function declaration. Thus, don't unwrap but just bail out (`find_param_with_region` is only applicable to functions).